runtime/compiler/optimizer/TreeLowering.cpp

```diff
@@ -1569,7 +1578,7 @@ TR::TreeLowering::lowerValueTypeOperations(TransformationManager& transformation
          {
          transformations.addTransformation(getTransformer<NonNullableArrayNullStoreCheckTransformer>(), node, tt);
          }
-      else if (node->getSymbolReference()->getReferenceNumber() == TR_ldFlattenableArrayElement)
+      else if (node->getSymbolReference() == symRefTab->findOrCreateLoadFlattenableArrayElementSymbolRef())
```

Comment:
I've added calls to `findOrCreate...` methods myself to compare something with the result of `TR::Node::getSymbolReference()` to check for a helper call, and I've seen lots of code in the JIT that does the same thing.  However, I'm starting to feel that it might be better to avoid creating the helper symbol reference unnecessarily.  In other words, maybe you don't need this change or the one that checks for `TR_strFlattenableArrayElement` below.

``` 
node->getSymbolReference()->getReferenceNumber() == TR_helperReferenceNumber
```

Reply:
Addressed in 446e692